fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java

```diff
@@ -634,6 +642,41 @@ public void initFuzzyModeVariables() {
         this.enableShareHashTableForBroadcastJoin = random.nextBoolean();
     }
 
+    /**
+     * syntax:
+     * all -> use all event
+     * all except event_1, event_2, ..., event_n -> use all events excluding the event_1~n
+     * event_1, event_2, ..., event_n -> use event_1~n
+     */
+    @VariableMgr.VarAttr(name = NEREIDS_TRACE_EVENT_MODE)
+    public String nereidsTraceEventMode = "all";
+
+    private Set<Class<? extends Event>> parsedNereidsEventMode = EventSwitchParser.parse(Lists.newArrayList("all"));
+    private String cachedNereidsTraceEventMode = "all";
+
+    public void setEnableNereidsTrace(boolean enableNereidsTrace) {
+        this.enableNereidsTrace = enableNereidsTrace;
+    }
+
+    public void setNereidsTraceEventMode(String nereidsTraceEventMode) {
+        this.nereidsTraceEventMode = nereidsTraceEventMode;
+    }
+
+    public Set<Class<? extends Event>> getParsedNereidsEventMode() {
```

Comment:
it is still has the problem that user can execute statement like:
```sql
set nereids_trace_event_mode=xxxxx;
```
and cannot reveive any error message